linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,223 @@
+                for (; index > 0 && fast != null; index--) {
+                    slow = fast;
+                    fast = fast.next;
+                }
+                var node = new Node<>(value, slow, fast);
+                slow.next = node;
+                fast.prev = node;
+                size++;
+            }
+        }
+    }
+
+    /**
+     * Inserts the specified element at the beginning of this list.
+     *
+     * @param value element to be inserted
+     */
+    public void addFirst(T value) {
+        var node = new Node<>(value, null, head);
+        if (head != null) {
+            head.prev = node;
+            head = node;
+        } else {
+            head = tail = node;
+        }
+        size++;
+    }
+
+    /**
+     * Appends the specified element to the end of this list.
+     *
+     * @param value element to be appended
+     */
+    public void addLast(T value) {
+        var node = new Node<>(value, tail, null);
+        if (tail != null) {
+            tail.next = node;
+            tail = node;
+        } else {
+            tail = head = node;
+        }
+        size++;
+    }
+
+    /**
+     * Removes and returns the first element from this list.
+     *
+     * @return the value of the removed first element
+     * @throws NoSuchElementException if the list is empty
+     */
+    public T removeFirst() {
+        if (head == null) {
+            throw new NoSuchElementException();
+        }
+        var value = head.value;
+        head = head.next;
+        head.prev = null;
+        size--;
+        return value;
+    }
+
+    /**
+     * Removes and returns the last element from this list.
+     *
+     * @return the value of the removed last element
+     * @throws NoSuchElementException if the list is empty
+     */
+    public T removeLast() {
+        if (tail == null) {
+            throw new NoSuchElementException();
+        }
+        var value = tail.value;
+        tail = tail.prev;
+        tail.next = null;
+        size--;
+        return value;
+    }
+
+    /**
+     * Removes and returns the element at the specified position in this list.
+     * For index == 0 this is equivalent to removeFirst(); for index == size - 1 — removeLast().
+     *
+     * @param index the index of the element to be removed (0..size-1)
+     * @return the value of the removed element
+     * @throws IndexOutOfBoundsException if the index is out of range
+     */
+    public T remove(int index) {
+        if (!checkElementIndex(index)) {
+            throw new IndexOutOfBoundsException();
+        } else if (index == 0) {
+            return removeFirst();
+        } else if (index == size - 1) {
+            return removeLast();
+        } else {
+            Node<T> slow = null;
+            Node<T> fast = head;
+            for (; index > 0 && fast != null; index--) {
+                slow = fast;
+                fast = fast.next;
+            }
+            slow.next = fast.next;
```

Comment:
it updates slow.next = fast.next but doesn't update the previous reference of the next node. This breaks the doubly-linked structure.